app/application.py

```diff
@@ -1069,6 +1094,61 @@ def clean_up(tmp_dir):
             raise
 
 
+def save_local_json(json_data, filename):
+    logger = logging.getLogger(__app_name__)
+    logger.info("Saving Locally...")
+    time_in_str = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+    base_filename, _ = os.path.splitext(filename)
+    file_path = os.path.join(
+        "tmp/", base_filename + "_" + time_in_str + ".json"
+    )
+    with open(file_path, "w") as json_file:
+        json.dump(json_data, json_file, indent=4)
+    logger.info(f"Model stored at path {file_path}")
+    return file_path
+
+
+def generate_srt(data, filename):
+    logger = logging.getLogger(__app_name__)
+    logger.info("Saving Locally...")
+    time_in_str = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+    base_filename, _ = os.path.splitext(filename)
+    output_file = os.path.join(
+        "tmp/", base_filename + "_" + time_in_str + ".srt"
+    )
+    if not os.path.isdir("/".join(output_file.split("/")[:-1])):
+        os.makedirs("/".join(output_file.split("/")[:-1]))
+    with open(output_file, "w") as f:
+        for index, segment in enumerate(data):
+            start_time, end_time, text = segment
+            f.write(f"{index+1}\n")
+            f.write(f"{format_time(start_time)} --> {format_time(end_time)}\n")
+            f.write(f"{text.strip()}\n\n")
+    logger.info("File saved")
+    return output_file
+
+
+def format_time(time):
+    hours = int(time / 3600)
+    minutes = int((time % 3600) / 60)
+    seconds = int(time % 60)
+    milliseconds = int((time % 1) * 1000)
+    return f"{hours:02d}:{minutes:02d}:{seconds:02d},{milliseconds:03d}"
+
+
+def upload_file_to_s3(file_path):
+    s3 = boto3.client("s3")
+    config = dotenv_values(".env")
+    bucket = config["S3_BUCKET"]
```

Comment:
not sure what permissions are assumed here but this didn't upload:

```
> tstbtc './test/testAssets/test_video.mp4' test-dir -p -u -V
Pytube Error: 'list'
2023-06-05 16:19:40,907 - tstbtc - INFO - Transcribing video: test_video.mp4
2023-06-05 16:19:40,907 - tstbtc - INFO -
        This tool will convert Youtube videos to mp3 files and then transcribe         them to text using Whisper.

2023-06-05 16:19:40,907 - tstbtc - DEBUG - Initializing FFMPEG...
2023-06-05 16:19:40,907 - tstbtc - DEBUG - Initialized FFMPEG
2023-06-05 16:19:41,026 - tstbtc - INFO - Converting video to mp3... Please wait.
2023-06-05 16:19:41,026 - tstbtc - INFO - /Users/jonas/chaincode/yt2btc/test/testAssets/test_video.mp3
MoviePy - Writing audio in /var/folders/4x/qwmmyd9n11q6cvwbqpxwk3r40000gn/T/tmp75_8wsv2/test_video.mp3
MoviePy - Done.
2023-06-05 16:19:41,369 - tstbtc - INFO - Converted video to mp3
2023-06-05 16:19:41,369 - tstbtc - INFO - Transcribing audio to text using whisper ...
2023-06-05 16:19:43,892 - tstbtc - INFO - Saving Locally...
2023-06-05 16:19:43,893 - tstbtc - INFO - File saved
ERROR:root:Error uploading file to S3 bucket: Unable to locate credentials
INFO:root:Removed video and audio files
2023-06-05 16:19:46,131 - tstbtc - INFO - Creating markdown file
INFO:tstbtc:Creating markdown file
2023-06-05 16:19:46,131 - tstbtc - INFO - writing .md file
INFO:tstbtc:writing .md file
2023-06-05 16:19:46,131 - tstbtc - INFO - wrote .md file
INFO:tstbtc:wrote .md file
```
(and no files appear to have been saved locally)

Reply:
> ERROR:root:Error uploading file to S3 bucket: Unable to locate credentials
This issue arises because probably `aws-cli` hasn't been setup properly. Kindly take a look at [this](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-quickstart.html) (try generating the short-term credentials for the same). I'll create a separate section in README.md for it.
- The files are temporarily stored, which is handled by the library.